lib/overcommit/utils.rb

```diff
@@ -124,15 +124,14 @@ def execute(args)
       # which we do not need to know the result.
       #
       # @param args [Array<String>]
-      # @return [Thread] thread watching the resulting child process
+      # @return [ChildProcess] thread watching the resulting child process
       def execute_in_background(args)
         if args.include?('|')
           raise Overcommit::Exceptions::InvalidCommandArgs,
                 'Cannot pipe commands with the `execute_in_background` helper'
         end
 
-        # Dissociate process from parent's input/output streams
-        Process.detach(Process.spawn({}, *args, [:in, :out, :err] => '/dev/null'))
+        Subprocess.spawn_in_background(args)
```

Comment:
How does this deal with long-running background processes that continue running after the hook run has completed?

The reason we used `Process.detach` here was to ensure that we wouldn't create a [zombie process](http://en.wikipedia.org/wiki/Zombie_process). I'm not sure how `ChildProcess` deals with this situation, but so far it hasn't been necessary to worry about because we don't use it for these sorts of background tasks&mdash;it's always been used in a context that blocks the parent Overcommit process until it completes.

Furthermore, I'm not sure what happens if terminating the parent process terminates the child process. We wouldn't want this to happen for hooks like `IndexTags` which can take many seconds to index large repositories.